Add web UI > Organizations content (#907)
giantswarm/docs: src/content/ui-api/web/organizations/access-control/index.md

---
linkTitle: Access control
title: Access control for organizations in the web user interface
description: Learn how to use our web UI for organization RBAC to quickly associate service accounts with organizations.
weight: 20
menu:
  main:
    identifier: orgs-accesscontrol
    parent: web-orgs
owner:
  - https://github.com/orgs/giantswarm/teams/biscuit
user_questions:
  - How can I inspect roles and permissions bound to an org-namespace?
  - How can I create a service account in the web interface?
  - How can I bind a service account to an org-namespace in the web interface?
last_review_date: 2021-06-16
---

# Access control for organizations in the web user interface

Learn how to quickly grant and revoke access to an organization's resources via the web user interface, using Kubernetes role based access control (RBAC) mechanisms.

{{% organization_mgmt_mapi_disclaimer %}}

Granting access means that the users (when associating users or groups) or programs (when using service accounts) affected will be able to access resources via the Management API, including via the web user interface itself.

> The access control UI in the web interface provides quick access to a focused set of RBAC resources, with the goal to make it easy for you to give someone &ndash; be it individuals or groups &ndash; or service accounts access to resources in an organization's namespace.

Reviewer: ```suggestion
The access control UI in the web interface provides quick access to a focused set of RBAC resources, with the goal to make it easy for you to give individuals, groups, or service accounts access to resources in an organization's namespace.
```

does that read a little easier? i don't love the current construction, but it's hard to find something better

## Where to find it {#where-to-find}

> 1. Log in to the web user interface, using the URL matching your management cluster/installation.

Reviewer: > management cluster/installation

could we be a bit more specific here?

Author: Not yet. We have a little widget planned to help with this (https://github.com/giantswarm/giantswarm/issues/16876) but it's not there yet.

---
linkTitle: Access control
title: Access control for organizations in the web user interface

> description: Learn how to use our web UI for organization access control to quickly associate users, groups, and service accounts with organizations.

Reviewer: 'organization access control' is a bit of a mouthful - maybe something like 'for organization management'?

Author: Going for "organization RBAC", since "organization management" is more the broader context here.

weight: 20
menu:
  main:
    identifier: orgs-accesscontrol
    parent: web-orgs
owner:
  - https://github.com/orgs/giantswarm/teams/biscuit
user_questions:
  - How can I give someone access to an Organization in the web UI?
  - How can I give someone access to the web UI?
  - Where can I find the access control UI?
last_review_date: 2021-04-20
---

# Access control for organizations in the web user interface

Learn how to quickly grant and revoke access to an organization's resources via the web user interface, using Kubernetes role based access control (RBAC) mechanisms.

{{% organization_mgmt_mapi_disclaimer %}}

> Granting access means that the users (when associating users or groups) or programs (when using service accounts) affected will be able to access resources via the management API, including via the web user interface itself.

Reviewer 2: ```suggestion
Granting access means that the users (when associating users or groups) or programs (when using service accounts) affected will be able to access resources via the Management API, including via the web user interface itself.
```

Author: I'll have to get this sorted everywhere. It's not consistent currently. Thanks

> The access control UI in the web interface provides quick access to a focussed set of RBAC resources, with the goal to make it easy for you to give someone – be it individuals or groups – or service accounts access to resources in an organization's namespace.

Reviewer: ```suggestion
The access control UI in the web interface provides quick access to a focused set of RBAC resources, with the goal to make it easy for you to give someone – be it individuals or groups – or service accounts access to resources in an organization's namespace.
```

## Where to find it {#where-to-find}

1. Log in to the web user interface, using the URL matching your management cluster/installation.
2. Click **Organizations** in the main menu in the top.
3. Click the organization you want to manage access for.
4. Select the **Access control** tab.

## User interface overview {#ui-overview}

![Access control user interface overview](access-control-ui.png)

The access control user interface provides:

> - A list of roles, both from the organization's namespace and cluster roles, to select from. Each role defines a specific set of permissions for a number of resources. Cluster roles are marked using a 🌐 icon. In the list, it is also indicated how many users and groups are bound to each role in this organization's namespace.

Reviewer: > A list of roles, both from the organization's namespace and cluster roles, to select from.

what do you think about

> A list of roles to select from, sourced from both the organization's namespace and cluster roles

? to avoid the embedded clause (which i always find challenging as a reader, especially for technical content)

Author: Good!

- For the selected role:

    - Basic information regarding whether the role is a cluster role or a namespaced role, and who manages the role according to the `app.kubernetes.io/managed-by` label (defaulting to **You** if the label does not exist).

    - The list of **Subjects** associated with the role. Subjects are individual users or user groups, as defined in the identity provider you use for authentication.

    - The list of **Permissions** granted to any subject associated with the role.

> While the **Subjects** tab will be explained in more detail below under [granting access](#grantin-access), the information shown under **Permissions** deserves a closer look here.

Reviewer 2: ```suggestion
While the **Subjects** tab will be explained in more detail below under [granting access](#granting-access), the information shown under **Permissions** deserves a closer look here.
```

### Permissions

![Access control user interface overview](permissions.png)

The example above shows permission details for a role that grants:

- all permissions to `apps` and `charts` from the `application.giantswarm.io` API group.
- only `get`, `list`, and `watch` permission to `appcatalogs` and `appcatalogentries` from the `application.giantswarm.io` API group.
- only `get`, `list`, and `watch` permission for clusters.

> The single letter icons in the **Verbs** column each represent one of the seven standard verbs: `get`, `watch`, `list`, `create`, `update`, `patch`, and `delete`. The last items, showing an ellipsis `⋯`, stands for additional, arbitrary verbs. Hovering these icons with your pointer will reveal additional information in a tooltip.

Reviewer: ```suggestion
The single letter icons in the **Verbs** column each represent one of the seven standard verbs: `get`, `watch`, `list`, `create`, `update`, `patch`, and `delete`. The last items, showing an ellipsis `⋯`, stands for additional, arbitrary verbs. Hovering over these icons with your pointer will reveal additional information in a tooltip.
```

> The single letter icons in the **Verbs** column each represent one of the seven standard verbs: `get`, `watch`, `list`, `create`, `update`, `patch`, and `delete`. The last items, showing an ellipsis `⋯`, stands for additional, arbitrary verbs. Hovering these icons with you pointer will reveal additional information in a tooltip.

Reviewer: ```suggestion
The single letter icons in the **Verbs** column each represent one of the seven standard verbs: `get`, `watch`, `list`, `create`, `update`, `patch`, and `delete`. The last items, showing an ellipsis `⋯`, stands for additional, arbitrary verbs. Hovering these icons with your pointer will reveal additional information in a tooltip.
```

> The single letter icons in the **Verbs** column each represent one of the seven standard verbs: `get`, `watch`, `list`, `create`, `update`, `patch`, and `delete`. The last items, showing an ellipsis `⋯`, stands for additional, arbitrary verbs. Hovering over these icons with your pointer will reveal additional information in a tooltip.

Reviewer 3: ```suggestion
The single letter icons in the **Verbs** column each represent one of the seven standard verbs: `get`, `watch`, `list`, `create`, `update`, `patch`, and `delete`. The last items, showing an ellipsis `⋯`, stands for additional, arbitrary verbs. Hovering over these icons with your pointer will reveal additional information in a tooltip.
```
I'm not sure what arbitrary means here. I'm pretty sure you didn't mean random here. Maybe we can find a less arbitrary word 😉

Author: Kubernetes allows to use any string as a "verb" for permissions. The verbs are meant as in `<subject> can to <verb> to <object>`. It is completely up to the user to invent things like `run`, `check` or `banana`.

- Show all roles with `read` in the name.
- Show all roles where `someone@example.com` appears as a subject.
- Show all roles explicitly granting permission to `events`.

## Granting access {#granting-access}

Granting access to an organization's resources means, in Kubernetes RBAC terms: you assign a role or cluster role to the selected organization's namespace, by creating role bindings in this namespace.

In order to assign a role to a user, group, or service account, follow these steps:

### 1. Select a role

Find and select the role or cluster role to assign within the list on the left hand side. Select the role that defines the permissions you want to grant.

For example, the `cluster-admin` ClusterRole, when assigned here, will grant all permissions within this organization's namespace. The `read-all` ClusterRole instead gives read-only access to all resources in this namespace.

### 2. Double-check permissions

By inspecting the _Permissions_ tab, double check that the role grants only the permissions you are expecting.

### 3. Add subjects

On the _Subjects_ tab, add the group(s) or user(s) you want to assign to the selected role.

> When adding a **group** or a **user**, you have to enter the exact user or group identifier as your identity provider presents them. This includes upper- and lowercase spelling. User identifiers may or may not be email addresses, according to your set-up.

Reviewer: > upper-

what's with the hypen here?

Author: Oh, my bad transfer from German language.

Reviewer: ezzz

> When adding a **group** or a **user**, you have to enter the exact user or group identifier as your identity provider presents them. This includes upper- and lowercase spelling. User identifiers may or may not be email addresses, according to your set-up.

Author: ```suggestion
When adding a **group** or a **user**, you have to enter the exact user or group identifier as your identity provider presents them. This includes upper and lowercase spelling. User identifiers may or may not be email addresses, according to your set-up.
```

> **Note:** Your Giant Swarm solution engineer (SE) is happy to assist in case there are uncertainties about the correct spelling.

Reviewer: ```suggestion
**Note:** Your Giant Swarm account engineer (AE) is happy to assist in case there are uncertainties about the correct spelling.
```

> **Note:** Your Giant Swarm account engineer (AE) is happy to assist in case there are uncertainties about the correct spelling.

Reviewer 3: I'd switch 'spelling' with 'format', since I don't think the actual spelling is the problem here. I'm under the impression that it's more about case usage, etc.

Author: Will change

> You can enter or paste multiple identifiers, separated by blank or comma, into a single input field.

Reviewer: ```suggestion
You can enter or paste multiple identifiers, separated by a space or comma, into a single input field.
```

When adding a **service account** as a subject, you are free to select from the existing ones in the current organization's namespace, by using the suggestions in the input field. When typing some letters, these suggestions will present existing service accounts that match your input. However you can also create new service accounts here by entering new names here.

In either case, confirm by clicking the _OK_ button.

Done. You have successfully assigned subjects to one specific role for one specific organization.

### Resulting resources

As a result of the action above, there will be new role bindings (`RoleBinding` resources) in the organization's namespace. These resources follow the naming convention

```nohighlight
<role-name>-<timestamp>
```

As an example, a `RoleBinding` for a `Role` or `ClusterRole` named `read-all` could be named `read-all-1618868398846`.

As mentioned, new service accounts may have been created in the organization's namespace.

You are free to edit or delete the role bindings and service accounts created via the web user interface in some other way via the management API. Treat them as if they were yours.

> ## Revoking access

Reviewer 4: I am putting myself on customer shoes and I see they would like to remove a user from a certain group. I know this means removing the user from IdP but I would add a warning/note here explaining the case

Reviewer 4: Maybe the same for the addition of a certain user to a group in IdP (like "adding a user in the same user group on your IdP connector will result in assigning permissions automatically without any further action in our platform")

Author: Sounds reasonable, will add something.
